src/main/java/jdbc/web/filter/UserFilter.java

```diff
@@ -24,33 +38,36 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
         HttpServletRequest req = (HttpServletRequest) servletRequest;
         HttpServletResponse res = (HttpServletResponse) servletResponse;
         String path = req.getServletPath() + req.getPathInfo();
-        
-        if (req.getSession().getAttribute("LOGGED_USER") != null) {
-            req.getRequestDispatcher("/WEB-INF/views/home.jsp")
-                    .forward(req, res);
-            return;
-        }
 
         Cookie[] cookies = req.getCookies();
         String token = null;
-        for (Cookie c : cookies) {
-            if (c.getName().equals("MATE")) {
-                token = c.getValue();
+        if (cookies != null) {
+            for (Cookie c : cookies) {
+                if (c.getName().equals("MATE")) {
+                    token = c.getValue();
+                }
             }
         }
 
         if (token == null) {
-            if (path.equals("/servlet/login") || path.equals("/servlet/register")) {
+            if (path.equals("/servlet/login")
+                    || path.equals("/servlet/register")
+                    || path.equals("/servlet/403")) {
```

Comment:
should be only registration and login without 403
for errors should be different method "processDenied" which should redirect to error pages